From d12e39f1282f1da0f7198b06484b1d68bbbd5fde Mon Sep 17 00:00:00 2001
From: Pablo Fraile <socdelbarca72@gmail.com>
Date: Fri, 15 Sep 2017 15:06:10 +0100
Subject: [PATCH] Use OMX.google.h264 encoder (Thanks to Hikari No Tenshi)

This will fix video recording on mtk devices, that use the h264 google encoder, like for example the BQ Aquaris E4.5 (krillin)
---
 app/src/main/java/org/lineageos/recorder/screen/EncoderDevice.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/src/main/java/org/lineageos/recorder/screen/EncoderDevice.java b/app/src/main/java/org/lineageos/recorder/screen/EncoderDevice.java
index 8325826..aced779 100644
--- a/app/src/main/java/org/lineageos/recorder/screen/EncoderDevice.java
+++ b/app/src/main/java/org/lineageos/recorder/screen/EncoderDevice.java
@@ -272,7 +272,7 @@ public void start(Attributes attributes) {
         // create a surface from the encoder
         Log.i(LOGTAG, "Starting encoder at " + width + "x" + height);
         try {
-            venc = MediaCodec.createEncoderByType("video/avc");
+            venc = MediaCodec.createByCodecName("OMX.google.h264.encoder");
         } catch (IOException e) {
             Log.wtf(LOGTAG, "Can't create AVC encoder!", e);
         }
